nodejs/awsx/apigateway/cognitoauthorizer.ts

```diff
@@ -0,0 +1,146 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// These APIs are currently experimental and may change.
+
+import * as aws from "@pulumi/aws";
+import * as pulumi from "@pulumi/pulumi";
+
+/**
+ * CognitoAuthorizer provides the definition for a Cognito User Pools Authorizer for API Gateway.
+ */
+export interface CognitoAuthorizer {
+    /**
+     * The name for the Authorizer to be referenced as. This must be unique for each unique
+     * authorizer within the API. If no name if specified, a name will be generated for you.
+     */
+    authorizerName?: string;
+
+    /**
+     * parameterName is the name of the header containing the authorization token.
+     * */
+    parameterName: string;
+
+    /**
+     * Defines where in the request API Gateway should look for identity information. The value must
+     * be "header" for a Cognito Authorizer.
+     */
+    parameterLocation: "header";
```

Comment:
as this is a public API, this is just something onerous for the user.  We should remove and do this on their behalf.

Reply:
oh.  i think i see.  this is *not* what the user wuld create.  instead, they woudl use the 'args' below to get one of these?  in that case, this is likely fine.

Reply:
Yes the hope is they use the helper function but could also choose to define a `CognitoAuthorizer` directly. I will take these out and just add them where I create the SwaggerCognitoAuthorizer.